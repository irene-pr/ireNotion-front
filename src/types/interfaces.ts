export interface IChecklist {
  isChecked: boolean;
  paragraph: string;
}

export interface INote {
  type: string;
  color: string;
  position?: number;
  title?: string;
  paragraph?: string;
  list?: Array<string | IChecklist>;
  id: string;
}

export interface IBoard {
  type: string;
  name: string;
  notes: Array<INote>;
  id: string;
}

export interface IUserContent {
  name: string;
  boards: Array<IBoard>;
}

export interface IUserData {
  userId: string;
  username: string;
}

export interface IState {
  isLoggedIn: boolean;
  userData: IUserData;
  userContent: IUserContent;
}

export interface IUserLoginData {
  username: string;
  password: string;
}

export interface IUserRegisterData {
  name: string;
  username: string;
  password: string;
}
