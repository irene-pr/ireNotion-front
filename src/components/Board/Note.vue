<template>
  <article
    class="note note-paragraph"
    :note="note"
    v-if="note.type === 'paragraph'"
    :class="note.color"
    @dblclick="onClickOpenModal"
  >
    <font-awesome-icon
      icon="pencil-alt"
      class="form__icon-update"
      @click="onClickOpenModal"
    ></font-awesome-icon>
    <font-awesome-icon
      icon="times"
      class="form__icon-close"
      @click="onClickDelete"
    ></font-awesome-icon>
    <h4 v-if="note.title !== ''">{{ note.title }}</h4>
    <p v-if="note.paragraph !== ''">{{ note.paragraph }}</p>
  </article>
  <article
    class="note note-image"
    :note="note"
    v-if="note.type === 'image'"
    :class="note.color"
  >
    <font-awesome-icon
      icon="times"
      class="form__icon-close"
    ></font-awesome-icon>
    <img :src="note.file" alt="" />
  </article>
</template>

<script>
import { defineComponent } from "vue";
import { mapActions } from "vuex";

export default defineComponent({
  name: "Note",
  props: ["note", "idBoard"],
  methods: {
    ...mapActions([
      "deleteNote",
      "updateNote",
      "setUpdateNoteModal",
      "setIdForModal",
    ]),
    onClickDelete() {
      const params = `${this.idBoard}/${this.note.id}`;
      this.deleteNote(params);
    },
    onClickOpenModal() {
      this.setUpdateNoteModal(true);
      this.setIdForModal(this.note.id);
    },
  },
});
</script>

<style lang="scss" scoped>
@import "@/styles/_variables";

.note {
  * {
    margin: 0;
  }

  position: relative;
  width: 100%;
  padding: 25px;
  margin: 10px 0;
  display: flex;
  flex-direction: column;
  font-family: $Lato;
  box-shadow: 0px 5px 7px -4px black;

  @media only screen and (max-width: 480px) {
    width: 90%;
  }

  &-paragraph {
    word-wrap: break-word;
    h4 {
      padding: 0 0 10px 10px;
      font-family: $hand-writing;
      font-weight: bold;
      font-size: 26px;
    }
    p {
      font-family: $hand-writing;
      font-size: 18px;
      font-weight: 100;
    }
  }
  &-image {
    padding: 0;
    img {
      object-fit: contain;
    }
  }
  &:hover {
    .form__icon-close {
      display: block;
    }
    .form__icon-update {
      display: block;
    }
  }
}
.form__icon-close {
  display: none;
  position: absolute;
  right: 10px;
  top: 10px;
  @media only screen and (max-width: 720px) {
    display: block;
  }
}
.form__icon-update {
  display: none;
  position: absolute;
  left: 10px;
  top: 10px;
  font-size: 12px;
  @media only screen and (max-width: 720px) {
    display: block;
  }
}
.orange {
  background-color: $note-orange;
  color: black;
}
.blue {
  background-color: $note-blue;
  color: white;
}
.pink {
  background-color: $note-pink;
  color: white;
}
.yellow {
  background-color: $note-yellow;
  color: black;
}
.green {
  background-color: $note-green;
  color: black;
}
</style>
