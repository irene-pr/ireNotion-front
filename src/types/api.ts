import { INote } from "./store";

export interface IUserLoginData {
  username: string;
  password: string;
}

export interface IUserRegisterData {
  name: string;
  username: string;
  password: string;
}

export interface IBodyEditNameBoard {
  idBoard: string;
  newName: string;
}
export interface IBodyUpdateNote {
  idNote: string;
  updatedNote: INote;
}
