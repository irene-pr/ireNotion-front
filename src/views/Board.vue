<template>
  <main class="board-page">
    <h1>This is the Board Page!</h1>
  </main>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "Login",
  components: {},
});
</script>

<style lang="scss">
@import "@/styles/_variables";
</style>
