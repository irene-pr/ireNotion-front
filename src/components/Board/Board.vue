<template>
  <section class="board">
    <div class="board-header">
      <h3 class="board-header__header">{{ board.name }}</h3>
      <div class="board-header__buttons">
        <button class="board-header__button--note">Add Note</button>
        <button class="board-header__button--image">Add Image</button>
      </div>
    </div>

    <draggable class="board__draggable-area" :board="board" @change="log">
      <Note v-for="note in board.notes" :note="note" :key="note.id" />
    </draggable>
  </section>
</template>

<script>
import { defineComponent } from "vue";
import { VueDraggableNext } from "vue-draggable-next";
import Note from "@/components/Board/Note.vue";
import { IBoard } from "@/types/interfaces";

export default defineComponent({
  name: "Board",
  components: {
    draggable: VueDraggableNext,
    Note,
  },
  props: { board: IBoard },
  data() {
    return {
      enabled: true,
      dragging: false,
    };
  },
  methods: {
    log(event) {
      console.log(event);
    },
  },
});
</script>
<style lang="scss" scoped>
.board {
  max-width: 100%;
  width: min-content;
  padding: 15px;
  border: 2px red dotted;
  &__draggable-area {
    width: min-content;
    padding: 15px;
    border: 2px magenta dotted;
  }
  .board-header {
    display: flex;
    justify-content: space-between;
    &__header {
      margin: 0;
      font-family: "Fredericka the Great", cursive;
      font-weight: 100;
      font-size: 28px;
      padding-left: 20px;
    }
    &__buttons {
      display: flex;
      align-content: center;
    }
    &__button {
    }
  }
}
</style>
