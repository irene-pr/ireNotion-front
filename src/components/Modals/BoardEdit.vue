<template>
  <div
    class="modal-edit-board"
    :class="'modal-edit-board--' + themeSurfaces"
    @click="closeModal"
  >
    <div class="board-edit-modal" :class="'board-edit-modal--' + themeSurfaces" @click.stop>
      <font-awesome-icon
        icon="times"
        class="form__icon-close"
        @click="closeModal"
      ></font-awesome-icon>
      <h2 class="board-edit-modal__title">Edit the board name</h2>
      <form
        class="board-edit-form"
        @submit.prevent="onSubmit"
        autocomplete="off"
      >
        <div class="board-edit-form__name">
          <label class="board-edit-form__label" for="newBoardName">Name</label>
          <input
            class="board-edit-form__input"
            type="text"
            id="newBoardName"
            v-model="name"
            placeholder="New board name"
          />
        </div>
        <button
          class="board-edit-form__button"
          type="submit"
          :class="'board-edit-form__button--' + themeSurfaces"
          :disabled="name === ''"
        >
          Edit
        </button>
      </form>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { mapActions, mapState } from "vuex";
import { IBoard } from "@/types/interfaces";

export default defineComponent({
  name: "BoardEdit",
  data() {
    return {
      name: "",
    };
  },
  computed: {
    ...mapState(["userContent", "themeSurfaces", "idForModal"]),
  },
  methods: {
    ...mapActions(["editNameBoard", "setBoardEditModal", "setIdForModal"]),
    onSubmit() {
      this.editNameBoard({ idBoard: this.idForModal, newName: this.name });
      this.closeModal();
    },
    closeModal() {
      this.setBoardEditModal(false);
      this.setIdForModal("");
    },
  },
  mounted() {
    this.name = this.userContent.boards.filter(
      ({ id }: IBoard) => id === this.idForModal
    )[0].name;
  },
});
</script>

<style lang="scss">
@import "@/styles/_variables";
@import "@/styles/_extends";

.modal-edit-board {
  position: fixed;
  width: 100%;
  height: 100%;
  z-index: 20;
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: $Lato;

  &--day-mode {
    background-color: rgba(255, 255, 255, 0.5);

    input {
      color: $theme-dark-color;
    }
  }
  &--night-mode {
    background-color: rgba(0, 0, 0, 0.5);

    input {
      color: $theme-light-color;
    }
  }
}

.board-edit-modal {
  position: relative;
  width: 300px;
  height: 200px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  border-radius: 8px;

  &--day-mode {
    background-color: $theme-light-color;
    color: $theme-dark-color;
    border: 2px solid $theme-dark-color-opacity;
  }
  &--night-mode {
    background-color: $theme-dark-color;
    color: $theme-light-color;
    border: 2px solid $theme-light-color-opacity;
  }

  &__title {
    margin-bottom: 0;
    font-weight: 400;
  }
  .board-edit-form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    &__name {
      margin: 25px 0;
    }
    &__label {
      padding-right: 20px;
    }
    &__input {
      height: 30px;
      padding-bottom: 0;
      border: none;
      border-bottom: 1px solid $theme-form-line;
      background-color: transparent;
      font-size: 14px;

      &:focus {
        outline: none;
        color: $theme-pink;
      }
    }

    &__button {
      @extend %button--little;
      margin-bottom: 10px;
      &--day-mode {
        background-color: rgba(255, 255, 255, 0.5);
        color: $theme-dark-color;
        border: 1px solid $theme-dark-color;
      }
      &--night-mode {
        background-color: rgba(0, 0, 0, 0.5);
        color: $theme-light-color;
        border: 1px solid $theme-light-color;
      }
    }
  }
  .form__icon-close {
    position: absolute;
    top: 10px;
    right: 10px;
  }
}
</style>
