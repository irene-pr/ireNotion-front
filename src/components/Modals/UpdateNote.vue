<template>
  <div class="modal" :class="'modal--' + themeSurfaces" @dblclick="closeModal">
    <div
      class="note-update-modal"
      :class="'note-update-modal--' + themeSurfaces"
    >
      <font-awesome-icon
        icon="times"
        class="form__icon-close"
        @click="closeModal"
      ></font-awesome-icon>
      <h2 class="note-update-modal__title">Edit the note</h2>
      <p
        class="note-update-modal__tip"
        :class="'note-update-modal__tip--' + themeSurfaces"
      >
        tip: you can leave empty fields, they won't leave an empty space.
      </p>
      <form
        class="note-update-form-paragraph"
        @submit.prevent="onSubmitParagraph"
        autocomplete="off"
        v-if="this.note.type === 'paragraph'"
      >
        <div class="note-update-form-paragraph__title">
          <label class="note-update-form-paragraph__label" for="title"
            >Title</label
          >
          <input
            class="note-update-form-paragraph__input"
            type="text"
            id="title"
            v-model="note.title"
            placeholder="New title"
          />
        </div>
        <div class="note-update-form-paragraph__text">
          <label class="note-update-form-paragraph__label" for="text"
            >Text
          </label>
          <textarea
            class="note-update-form-paragraph__input"
            type="text"
            id="text"
            v-model="note.paragraph"
            placeholder="New text"
          />
        </div>
        <p>Color</p>
        <div class="note-update-form-paragraph__color">
          <div class="radio-container">
            <label class="note-update-form-paragraph__label" for="pink"
              >Pink</label
            >
            <input
              class="note-update-form-paragraph__input"
              type="radio"
              id="pink"
              value="pink"
              v-model="note.color"
            />
          </div>
          <div class="radio-container">
            <label class="note-update-form-paragraph__label" for="yellow"
              >Yellow</label
            >
            <input
              class="note-update-form-paragraph__input"
              type="radio"
              id="yellow"
              value="yellow"
              v-model="note.color"
            />
          </div>
          <div class="radio-container">
            <label class="note-update-form-paragraph__label" for="blue"
              >Blue</label
            >
            <input
              class="note-update-form-paragraph__input"
              type="radio"
              id="blue"
              value="blue"
              v-model="note.color"
            />
          </div>
          <div class="radio-container">
            <label class="note-update-form-paragraph__label" for="orange"
              >Orange</label
            >
            <input
              class="note-update-form-paragraph__input"
              type="radio"
              id="orange"
              value="orange"
              v-model="note.color"
            />
          </div>
        </div>

        <button
          class="note-update-form-paragraph__button"
          type="submit"
          :class="'note-update-form-paragraph__button--' + themeSurfaces"
          :disabled="note === ''"
        >
          Edit
        </button>
      </form>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { mapActions, mapState } from "vuex";
import { IBoard, INote } from "@/types/interfaces";

export default defineComponent({
  name: "UpdateNote",
  data() {
    return {
      note: {
        type: "",
        order: "",
        color: "",
        title: "",
        paragraph: "",
        list: [],
        file: "",
      },
    };
  },
  computed: {
    ...mapState(["userContent", "themeSurfaces", "idForModal"]),
  },
  methods: {
    ...mapActions([
      "updateNote",
      "setUpdateNoteModal",
      "setIdForModal",
      "getUserContent",
    ]),
    onSubmitParagraph() {
      const note = {
        color: this.note.color,
        title: this.note.title,
        paragraph: this.note.paragraph,
      };
      this.updateNote({ idNote: this.idForModal, updatedNote: note });
      this.closeModal();
    },
    closeModal() {
      this.setUpdateNoteModal(false);
      this.setIdForModal("");
    },
  },
  mounted() {
    this.note = {
      ...this.userContent.boards
        .filter(
          ({ notes }: IBoard) =>
            notes.filter(({ id }: INote) => id === this.idForModal).length !== 0
        )[0]
        .notes.filter(({ id }: INote) => id === this.idForModal)[0],
    };
  },
});
</script>

<style lang="scss">
@import "@/styles/_variables";
@import "@/styles/_extends";

.modal {
  position: fixed;
  width: 100%;
  height: 100%;
  z-index: 20;
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: $Lato;

  &--day-mode {
    background-color: rgba(255, 255, 255, 0.7);

    input {
      color: $theme-dark-color;
    }
    textarea {
      color: $theme-dark-color;
    }
  }
  &--night-mode {
    background-color: rgba(0, 0, 0, 0.7);

    input {
      color: $theme-light-color;
    }
    textarea {
      color: $theme-light-color;
    }
  }
}

.note-update-modal {
  position: relative;
  width: 300px;
  height: 500px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  border-radius: 15px;

  &--day-mode {
    background-color: $theme-light-color;
    color: $theme-dark-color;
    border: 2px solid $theme-dark-color-opacity;
  }
  &--night-mode {
    background-color: $theme-dark-color;
    color: $theme-light-color;
    border: 2px solid $theme-light-color-opacity;
  }

  &__title {
    margin-bottom: 0;
    font-weight: 400;
  }
  &__tip {
    padding: 0 40px;
    text-align: center;
    font-size: 13px;
    &--day-mode {
      color: $theme-dark-color-opacity;
    }
    &--night-mode {
      color: $theme-light-color-opacity;
    }
  }
  .note-update-form-paragraph {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    &__title {
      margin-bottom: 20px;
    }
    &__text {
      margin: 0;
      display: flex;
      justify-content: space-between;
      align-items: center;
      textarea {
        resize: none;
        height: 100px;
        width: 180px;
        border: 1px solid $theme-form-line;
        font-family: $Lato;
      }
      label {
        padding-right: 22px;
      }
    }
    p {
      color: $theme-pink;
    }
    &__color {
      margin: 0 0 25px 0;
      display: flex;
      flex-direction: column;

      .radio-container {
        height: 25px;
        display: flex;
        justify-content: space-between;
        align-items: center;
      }
    }
    &__label {
      padding-right: 20px;
    }
    &__input {
      height: 30px;
      padding-bottom: 0;
      border: none;
      border-bottom: 1px solid $theme-form-line;
      background-color: transparent;
      font-size: 14px;

      &:focus {
        outline: none;
        color: $theme-pink;
      }
    }

    &__button {
      @extend %button--little;
      &--day-mode {
        background-color: rgba(255, 255, 255, 0.5);
        color: $theme-dark-color;
        border: 1px solid $theme-dark-color;
      }
      &--night-mode {
        background-color: rgba(0, 0, 0, 0.5);
        color: $theme-light-color;
        border: 1px solid $theme-light-color;
      }
    }
  }

  .form__icon-close {
    position: absolute;
    top: 10px;
    right: 10px;
  }
}
</style>
