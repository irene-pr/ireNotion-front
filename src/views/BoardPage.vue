<template>
  <main class="board-page">
    <h1>This is the Board Page!</h1>
    <Board v-for="board in boards" :board="board" v-bind:key="board.id" />
  </main>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { mapActions, mapState } from "vuex";
import Board from "@/components/Board/Board.vue";

export default defineComponent({
  name: "BoardPage",
  components: { Board },
  data() {
    return {
      enabled: true,
      boards: [],
      dragging: false,
    };
  },
  computed: {
    ...mapState(["userContent"]),
  },
  methods: {
    ...mapActions(["getUserContent"]),
  },
  mounted() {
    this.getUserContent();
    this.boards = this.userContent.boards;
  },
});
</script>

<style lang="scss">
@import "@/styles/_variables";
</style>
