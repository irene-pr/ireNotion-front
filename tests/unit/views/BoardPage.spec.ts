import { mount } from "@vue/test-utils";
import { createStore } from "vuex";
import { IState } from "@/types/interfaces";
import BoardPage from "@/views/BoardPage.vue";
import Board from "@/components/Board/Board.vue";
import BoardEdit from "@/components/Modals/BoardEdit.vue";
import UpdateNote from "@/components/Modals/UpdateNote.vue";

describe("Given a BoardPage view", () => {
  let mockedState: IState;
  let toggleThemeMock = jest.fn();
  let getUserContentMock = jest.fn();
  let checkTokenMock = jest.fn();
  let logoutUserMock = jest.fn();
  let createBoardMock = jest.fn();
  let onClickLogoutMock = jest.fn();
  let onClickCreateBoardMock = jest.fn();
  let onClickToggleThemeMock = jest.fn();
  let redierectToHomeMock = jest.fn();
  let store = createStore({});
  let options = {
    data() {
      return {
        enabled: true,
        dragging: false,
      };
    },
    components: {
      Board,
      BoardEdit,
      UpdateNote,
    },
    global: {
      plugins: [store],
    },
    mocks: {
      $store: store,
    },
    props: {
      board: {
        type: "board",
        name: "tablero1",
        notes: [{ type: "paragraph", color: "yellow" }],
        id: "123",
      },
    },
  };
  function getOptions() {
    return {
      data() {
        return {
          enabled: true,
          dragging: false,
        };
      },
      components: {
        Board,
        BoardEdit,
        UpdateNote,
      },
      global: {
        plugins: [store],
      },
      mocks: {
        $store: store,
      },
      props: {
        board: {
          type: "board",
          name: "tablero1",
          notes: [{ type: "paragraph", color: "yellow" }],
          id: "123",
        },
      },
    };
  }
  function getOptionsWithMethods() {
    return {
      data() {
        return {
          enabled: true,
          dragging: false,
        };
      },
      components: {
        Board,
        BoardEdit,
        UpdateNote,
      },
      global: {
        plugins: [store],
      },
      methods: {
        onClickLogout: onClickLogoutMock,
        onClickCreateBoard: onClickCreateBoardMock,
        onClickToggleTheme: onClickToggleThemeMock,
      },
      mocks: {
        $store: store,
      },
      props: {
        board: {
          type: "board",
          name: "tablero1",
          notes: [{ type: "paragraph", color: "yellow" }],
          id: "123",
        },
      },
    };
  }

  beforeEach(() => {
    mockedState = {
      isLoading: false,
      isLoggedIn: true,
      userContent: {
        name: "Franny",
        boards: [
          {
            type: "board",
            name: "tablero1",
            notes: [{ type: "paragraph", color: "yellow" }],
            id: "123",
          },
          {
            type: "board",
            name: "tablero2",
            notes: [{ type: "paragraph", color: "yellow" }],
            id: "234",
          },
        ],
      },
      userData: {
        userId: "",
        username: "",
      },
      themeHeaders: "night-mode",
      themeSurfaces: "day-mode",
      isBoardEditModal: false,
      isUpdateNoteModal: false,
      idForModal: "",
    };
    toggleThemeMock = jest.fn();
    getUserContentMock = jest.fn();
    checkTokenMock = jest.fn();
    logoutUserMock = jest.fn();
    createBoardMock = jest.fn();
    onClickLogoutMock = jest.fn();
    onClickCreateBoardMock = jest.fn();
    onClickToggleThemeMock = jest.fn();
    redierectToHomeMock = jest.fn();
    store = createStore({
      state() {
        return mockedState;
      },
      actions: {
        toggleTheme: toggleThemeMock,
        getUserContent: getUserContentMock,
        checkToken: checkTokenMock,
        logoutUser: logoutUserMock,
        createBoard: createBoardMock,
      },
      getters: {
        redierectToHome: redierectToHomeMock,
      },
    });
  });
  describe("When it is rendered", () => {
    test("Then it renders", () => {
      options = getOptions();
      mount(BoardPage, options);
    });
    test("Then the it matches the snapshot", () => {
      options = getOptions();
      const wrapper = mount(BoardPage, options);
      expect(wrapper.element).toMatchSnapshot();
    });
    test("Then it calls the action checkToken", () => {
      options = getOptions();
      mount(BoardPage, options);
      expect(checkTokenMock).toHaveBeenCalled();
    });
    test("Then it calls the action getUserContent", () => {
      options = getOptions();
      mount(BoardPage, options);
      expect(getUserContentMock).toHaveBeenCalled();
    });
    test("Then it renders header with the name of the User", () => {
      options = getOptions();
      const wrapper = mount(BoardPage, options);
      const header = wrapper.find("h1");

      expect(header.text()).toBe("Welcome, Franny!");
    });
  });
  describe("When the state isLoggedIn is false", () => {
    test("Then the getter redirectHome is called", () => {
      mockedState = {
        isLoading: false,
        isLoggedIn: false,
        userContent: {
          name: "Franny",
          boards: [
            {
              type: "board",
              name: "tablero1",
              notes: [{ type: "paragraph", color: "yellow" }],
              id: "123",
            },
            {
              type: "board",
              name: "tablero2",
              notes: [{ type: "paragraph", color: "yellow" }],
              id: "234",
            },
          ],
        },
        userData: {
          userId: "",
          username: "",
        },
        themeHeaders: "night-mode",
        themeSurfaces: "day-mode",
        isBoardEditModal: false,
        isUpdateNoteModal: false,
        idForModal: "",
      };
      store = createStore({
        state() {
          return mockedState;
        },
        actions: {
          toggleTheme: toggleThemeMock,
          getUserContent: getUserContentMock,
          checkToken: checkTokenMock,
          logoutUser: logoutUserMock,
          createBoard: createBoardMock,
        },
        getters: {
          redierectToHome: redierectToHomeMock,
        },
      });

      options = getOptions();
      mount(BoardPage, options);

      expect(redierectToHomeMock).toHaveBeenCalled();
    });
  });
  describe("When the header is clicked", () => {
    test("Then the action toggleTheme is called", () => {
      options = getOptions();
      const wrapper = mount(BoardPage, options);
      const header = wrapper.find("h1");

      header.trigger("click");

      expect(toggleThemeMock).toHaveBeenCalled();
    });
  });
  describe("When the logout button is clicked", () => {
    test("Then the action logout is called", () => {
      options = getOptions();
      const wrapper = mount(BoardPage, options);
      const logoutButton = wrapper.find(".nav-bar__button--logout");

      logoutButton.trigger("click");

      expect(logoutUserMock).toHaveBeenCalled();
    });
    test("Then the redirectToHome getter is called", () => {
      options = getOptions();
      const wrapper = mount(BoardPage, options);
      const logoutButton = wrapper.find(".nav-bar__button--logout");

      logoutButton.trigger("click");

      expect(redierectToHomeMock).toHaveBeenCalled();
    });
  });
  describe("When the add board button is clicked", () => {
    test("Then the action logout is called", () => {
      options = getOptions();
      const wrapper = mount(BoardPage, options);
      const addBoardButton = wrapper.find(".nav-bar__button--new-board");

      addBoardButton.trigger("click");

      expect(createBoardMock).toHaveBeenCalled();
    });
  });
});
