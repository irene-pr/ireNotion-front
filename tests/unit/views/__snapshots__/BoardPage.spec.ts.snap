// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a BoardPage view When it is rendered Then the it matches the snapshot 1`] = `
<div
  data-v-app=""
>
  
  <!--v-if-->
  <!--v-if-->
  <nav
    class="nav-bar night-mode"
  >
    <h1
      class="nav-bar__header"
    >
       Welcome, Franny! 
    </h1>
    <div
      class="nav-bar__buttons"
    >
      <button
        class="nav-bar__button nav-bar__button--new-board"
      >
         New 
        <span>
          Board
        </span>
      </button>
      <button
        class="nav-bar__button nav-bar__button--logout"
      >
         LOG 
        <span>
          OUT
        </span>
      </button>
    </div>
  </nav>
  <main
    class="board-page day-mode"
  >
    <div
      class="board-array day-mode"
    >
      
      <section
        class="board"
      >
        <div
          class="board-header night-mode"
        >
          <div
            class="board-header__top"
          >
            <h2
              class="board-header__header"
            >
              tablero1
            </h2>
            <font-awesome-icon
              class="form__icon-edit"
              icon="pencil-alt"
            />
            <font-awesome-icon
              class="form__icon-close"
              icon="times"
            />
          </div>
          <div
            class="board-header__buttons"
          >
            <button
              class="board-header__button board-header__button--note"
              type="button"
            >
               Add Note 
            </button>
            <button
              class="board-header__button board-header__button--image"
              type="button"
            >
               Add Image 
            </button>
          </div>
        </div>
        <div
          board="[object Object]"
          class="board__draggable-area"
        >
          
          
          <article
            class="note note-paragraph yellow"
            note="[object Object]"
          >
            <font-awesome-icon
              class="form__icon-update"
              icon="pencil-alt"
            />
            <font-awesome-icon
              class="form__icon-close"
              icon="times"
            />
            <!--v-if-->
            <!--v-if-->
          </article>
          <!--v-if-->
          
          
        </div>
      </section>
      <section
        class="board"
      >
        <div
          class="board-header night-mode"
        >
          <div
            class="board-header__top"
          >
            <h2
              class="board-header__header"
            >
              tablero2
            </h2>
            <font-awesome-icon
              class="form__icon-edit"
              icon="pencil-alt"
            />
            <font-awesome-icon
              class="form__icon-close"
              icon="times"
            />
          </div>
          <div
            class="board-header__buttons"
          >
            <button
              class="board-header__button board-header__button--note"
              type="button"
            >
               Add Note 
            </button>
            <button
              class="board-header__button board-header__button--image"
              type="button"
            >
               Add Image 
            </button>
          </div>
        </div>
        <div
          board="[object Object]"
          class="board__draggable-area"
        >
          
          
          <article
            class="note note-paragraph yellow"
            note="[object Object]"
          >
            <font-awesome-icon
              class="form__icon-update"
              icon="pencil-alt"
            />
            <font-awesome-icon
              class="form__icon-close"
              icon="times"
            />
            <!--v-if-->
            <!--v-if-->
          </article>
          <!--v-if-->
          
          
        </div>
      </section>
      
    </div>
  </main>
  
</div>
`;
